tools/package-source.js

```diff
@@ -99,6 +99,93 @@ var splitConstraint = function (c) {
            constraint: parsed.constraintString || null };
 };
 
+// Given the text of a README.md file, excerpts the text between the first and
+// second heading.
+//
+// Specifically - if there is text between the document name, and the first
+// subheading, it will take that text. If there is no text there, and only text
+// after the first subheading, it will take that text. It won't look any deeper
+// than that (in case the user intentionally wants to leave the section blank
+// for some reason). Skips lines that start with an exclamation point.
+var getExcerptFromReadme = function (text) {
+  // Split into lines.
+  var fullText = text.split("\n");
+  if (! fullText) {
+    return "";
+  }
+
+  // The lines of the excerpt that we will return to the caller.
+  var excerpt = [];
+
+  // We want to take the data between the first and second headers, so it is
+  // important to keep track of how many headers we have passed.
+  var headerNum = 0;
+
+  // Counter and termination condition for the while loop.
+  var lineNum = 0;
+  var done = false;
+
+  // We want to scan until we have an excerpt, or decide that we are not getting
+  // one. If we are only taking the first five lines of a long README file, we
+  // want to avoid scanning all of it.
+  while (! done) {
+    var currentLine = fullText[lineNum];
+    var trimmedLine = currentLine.trim();
+
+    // According to "https://help.github.com/articles/markdown-basics/", a
+    // heading is denoted by a '#' symbol, though, I think, the line at the
+    // top (===) should also count as a header for our purposes.
+    var isHeader =
+      (trimmedLine[0] === "#") ||
+      (trimmedLine[0] === "=");
+
+    // Exclamation points indicate images, which are almost never inline. We get
+    // cleaner results if we skip those lines entirely.
+    var notText = (trimmedLine[0] === "!");
+
+    // Don't excerpt anything before the first header! It is probably the
+    // title. If we haven't started excerpting already, don't excerpt the empty
+    // line (it might just be an empty line between headers, not any
+    // text). Don't excerpt things that we know are not text.
+    if (! isHeader && (headerNum > 0) && ! notText &&
+        (! _.isEmpty(excerpt) || trimmedLine !== "")) {
+      // If we are currently in the middle of excerpting, continue doing that
+      // until we hit hit a header (and this is not a header). Otherwise, if
+      // this is text, we should begin to excerpt it.
+      excerpt.push(currentLine);
+    } else if (! _.isEmpty(excerpt) && isHeader) {
+      // We have been excerpting lines, and came across a header. That means
+      // that we are done.
+      done = true;
+    } else if (isHeader) {
+      // We told users that we are going to take whatever is between the first
+      // and second header, but I think it is not always entirely clear what
+      // that means, and we get better results if we are willing to scan a
```

Comment:
Yeah, I was referring to the case that you are mentioning.

I wanted some way for people to have a README.md, but opt out of having a longform description text. I might be overthinking that... (But we did tell 'between first and second header' to users and they understood what we meant and supported it, so I would feel slightly awkward changing that now without good reason. Not that there can't be a good reason!)